### PR TITLE
archlinux: Use hostnamectl to set the transient hostname

### PR DESCRIPTION
**Pull request by klausenbusk**
## Proposed Commit Message
```
archlinux: Use hostnamectl to set the transient hostname

hostname (inetutils) isn't installed per default on arch, so switch
to hostnamectl which is installed per default (systemd).
```

## Test Steps
This has been tested manually with the Arch Linux [cloud-init package](https://archlinux.org/packages/community/any/cloud-init/) (`arch.py` patched with the changes in this commit) and `inetutils` forcefully removed (`pacman -Rsddn inetutils`).

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by klausenbusk** [comment]
The CLA stuff is part of: https://github.com/canonical/cloud-init/pull/796

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Your commit message lists the reason for making the change, but can you add a comment to the method to explain why you're overriding the default behavior? Also, can you add similar logging and try/except behavior that the base method has?

**Comment by klausenbusk** [comment]
Fixed

File: cloudinit/distros/arch.py
Comment (TheRealFalcon) [inline cloudinit/distros/arch.py:143]:
```suggestion
        LOG.debug("Non-persistently setting the system hostname to %s",
                  hostname)
        try:
```

**Review by TheRealFalcon — COMMENTED** [review]
I suggested one more change inline. Should be good to go after that. I'll wait for your other PR to merge given that the CLA change is in that one.

**Review by TheRealFalcon — APPROVED** [review]
Thanks!

**Comment by TheRealFalcon** [comment]
Hey @klausenbusk , the only thing this PR is waiting on is the CLA signers file, which is added in #796 . If you need to take your time on #796 , feel free to move that commit over to this PR and I can land this one as soon as possible. If you'd rather just wait until the other PR is finished, that is fine as well.

**Comment by TheRealFalcon** [comment]
@klausenbusk , since this PR was close to getting auto-closed, and since you've already signed the CLA, I took the liberty of moving your addition to the github CLA signers file over to this PR and merging it. Thanks for the contribution!

**Pull request merged**
